modules/system/assets/ui/js/waiaria.js

```diff
@@ -0,0 +1,324 @@
+        whiteList(event.key);
+
+        var $target = $(event.currentTarget);
+
+        // Set all social share buttons to false
+        $('#layout-sidenav a[role="button"]').attr('tabindex', '-1');
+
+        // Add wai-aria selected on the active tab
+        $target.attr('tabindex', '0');
+
+        if (event.key === 'ArrowUp') {
+            event.preventDefault();
+
+            var position = event.key === 'ArrowUp' ? 'first-child' : 'last-child',
+                $activated = $target.parent();
+
+            $activated.prev().children('#layout-sidenav a[role="button"]').click().focus();
+        } else if (event.key === 'ArrowDown') {
+            event.preventDefault();
+
+            var position = event.key === 'ArrowDown' ? 'first-child' : 'last-child',
+                $activated = $target.parent();
+
+            $activated.next().children('#layout-sidenav a[role="button"]').click().focus();
+        } else if (event.key === 'Home') {
+            event.preventDefault();
+
+            $('#layout-sidenav a[role="button"]').first().click().focus();
+        } else if (event.key === 'End') {
+            event.preventDefault();
+
+            $('#layout-sidenav a[role="button"]').last().click().focus();
+        }
+
+        // Important set to true to allow tab to enter and exit
+        return true;
+    });
+
+    /*
+     * Adds WAI-ARIA to breadcrumbs
+     */
+    if ($('.control-breadcrumb nav').not().length == 0) {
+        $('.control-breadcrumb ul').wrap('<nav>');
+        $('.control-breadcrumb nav').attr({'role': 'navigation', 'aria-label': 'breadcrumb'});
+        $('.control-breadcrumb li').last().attr('aria-current', 'page');
+        $('.control-breadcrumb li[aria-current="page"] a').attr('href', 'javascript:;');
+    }
+    if ($('.control-breadcrumb li[aria-current="page"] a').not().length == 0) {
+        $('.control-breadcrumb nav ul li').last().wrapInner('<a>');
+        $('.control-breadcrumb nav ul li a').last().attr('href', 'javascript:;');
+    }
+
+    /*
+     * Radio buttons
+     */
+    $('body').on('click keydown', '.custom-radio input[role="radio"]', function(event) {
+        // Run whitelist checker
+        whiteList(event.key);
+
+        var $target = $(event.currentTarget),
+            parentContainer = $target.closest('.radio-field');
+
+        // Remove WAI-ARIA checked and HTML checked attributes
+        $(parentContainer).find('input[role="radio"]').attr({
+            'aria-checked': 'false',
+            'tabindex': '-1',
+            'checked': false
+        });
+
+        // Add WAI-ARIA checked and HTML checked attributes to focus element
+        $(parentContainer).find('input[role="radio"]:checked').attr({
+            'aria-checked': 'true',
+            'tabindex': '0',
+            'checked': true
+        });
+    });
+
+    /*
+     * Slider switch buttons
+     */
+    $('body').on('change', '.switch-field input[role="checkbox"]', function () {
```

Comment:
@ayumihamsaki you lost `event` var here.

Reply:
@w20k thanks 👍